include/nfc/tnep/poller.h

```diff
@@ -0,0 +1,308 @@
+
+/**@brief Set NDEF API for the NFC TNEP Poller Device
+ *
+ * This function provides API for NDEF operation and indicates
+ * the NFC Device Tag Type. API can be changed in runtime but
+ * it must be set immediately after NFC Poller Device detects
+ * Tag Type during the Device Activation Activity.
+ *
+ * @param[in] api Pointer to the NDEF API structure.
+ * @param[in] tag_type NFC Device Tag Type.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int nfc_tnep_poller_api_set(const struct nfc_tnep_poller_ndef_api *api,
+			    enum nfc_tnep_tag_type tag_type);
+
+/**@brief Search if NDEF Message contains TNEP Service Parameters Records.
+ *
+ * Function for searching the TNEP Service Parameters Records in
+ * the given NDEF Message. NDEF Message has to be stored
+ * until all operations on services are finished.
+ *
+ * @param[in] ndef_msg The NDEF Message which can be the Initial TNEP Message.
+ * @param[out] param Pointer to structure where found service parameters will
+ *                   be stored.
+ * @param[in,out] cnt Count of service parameters which can be stored as an
+ *                    input. Count of found services parameters as an output.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int nfc_tnep_poller_svc_search(const struct nfc_ndef_msg_desc *ndef_msg,
+			       struct nfc_ndef_tnep_rec_svc_param *param,
+			       u8_t *cnt);
+
+/**@brief Select the TNEP Service.
+ *
+ * Function for selecting the given service. After service is selected
+ * NFC Poller waits a specific amount of time to perform the NDEF Read
+ * procedure to get new data from the NFC Tag device. This operation
+ * is asynchronous.
+ *
+ * @param[in] svc_buf Pointer to the buffer for new data from the NFC TNEP
+ *                    Tag Device. Buffer must be stored until the Select
+ *                    procedure is finished.
+ * @param[in] svc Pointer to The TNEP Service to select.
+ * @param[in] max_ndef_area_size Maximum size of NDEF Message.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int nfc_tnep_poller_svc_select(const struct nfc_tnep_buf *svc_buf,
+			       const struct nfc_ndef_tnep_rec_svc_param *svc,
+			       u32_t max_ndef_area_size);
+
+/**@brief Deselect the TNEP Service.
+ *
+ * Function for deselecting the TNEP service. Service can be deselected
+ * also by selecting the other TNEP service without calling this function.
+ *
+ * @retval 0 If the operation was successful.
+ *           Otherwise, a (negative) error code is returned.
+ */
+int nfc_tnep_poller_svc_deselect(void);
+
+/**@brief Read the TNEP Service data.
+ *
+ * Function for reading the TNEP Service data. The Service
+ * must be selected first. This operation is asynchronous.
+ *
+ * @param[in,out] svc_buf Pointer to buffer for data to read. It
```

Comment:
The parameter in declaration is of type `const struct nfc_tnep_buf *` can it realy be used as output parameter?

Reply:
I am not sure how to documented it because buf is constant but data is written to buf->data field.
Should it be only in parameter ?

Reply:
No let's leave it as _in/out_